### PR TITLE
MAINT: Switch away from nose in lib tests

### PR DESCRIPTION
**Pull request by DavidFair**
Nose is unsupported, and has broken with Python 3.11+, so switch away from this

### Special Notes:

---

### Submitter:

Have you (where applicable):

* [ ] Added unit tests?
* [ ] Checked the latest commit runs on Dev?
* [ ] Updated the example config file(s) and README?

---

### Reviewer

Does this PR:

* [ ] Place non-StackStorm code into the `lib` directory?
* [ ] Have unit tests for the action/sensor and `lib` layers?
* [ ] Have clear and obvious action parameter names and descriptions?

### CODE REVIEW

**Review by meoflynn — CHANGES_REQUESTED** [review]

File: lib/email_api/emailer.py [hunk 1]
```diff
@@ -73,34 +73,33 @@ def send_emails(self, emails: List[EmailParams]):
...
-                    self._build_email(email_params).as_string(),
+                    self.build_email_header(email_params).as_string(),
                 )
 
             logger.info(
                 "sending complete - time elapsed: %s seconds", time.time() - start
             )
 
-    def _build_email(self, email_params: EmailParams) -> MIMEMultipart:
+    def build_email_header(self, email_params: EmailParams) -> MIMEMultipart:
```
Comment (anish-mudaraddi) [inline lib/email_api/emailer.py:83]: Why did you change this? 
Technically this is not just building email header since it calls `build_email_body` in the method

File: tests/lib/test_openstack_security_groups.py [hunk 2]
```diff
@@ -115,42 +112,48 @@ def test_create_security_group_forwards_result(self):
...
-    def test_create_rule_throws_non_numeric(self, start_port, end_port):
+    def test_create_rule_throws_non_numeric(self):
         """
         Tests that non-numeric inputs throw
         """
         mocked_details = NonCallableMock()
-        mocked_details.port_range = (start_port, end_port)
-        with assert_raises(ValueError):
-            self.instance.create_security_group_rule(NonCallableMock(), mocked_details)
+        for start, end in (1, "a"), ("d", 2), ("*a", "a"), ("*", "*1"):
```
Comment (anish-mudaraddi) [inline tests/lib/test_openstack_security_groups.py:151]: why not use `pytest.mark.parametrize`? 
so we can see which one fails if the test fails?

File: tests/lib/test_openstack_security_groups.py [hunk 3]
```diff
@@ -115,42 +112,48 @@ def test_create_security_group_forwards_result(self):
...
-    def test_create_rule_throws_for_missing_port(self, start_port, end_port):
+    def test_create_rule_throws_for_missing_port(self):
         """
         Tests that missing port starting values throw
         """
         mocked_details = NonCallableMock()
-        mocked_details.port_range = (start_port, end_port)
-        with assert_raises(ValueError):
-            self.instance.create_security_group_rule(NonCallableMock(), mocked_details)
+        for start, end in (0, None), (None, 0), (None, None):
```
Comment (anish-mudaraddi) [inline tests/lib/test_openstack_security_groups.py:130]: why not use pytest.mark.parametrize?
so we can see which one fails if the test fails?

File: tests/lib/test_openstack_identity.py [hunk 4]
```diff
@@ -198,27 +198,26 @@ def test_delete_project_handles_resource_not_found(self):
...
-    @raises(MissingMandatoryParamError)
     def test_find_project_raises_missing_project_identifier(self):
         """
         Tests that a missing mandatory parameter is missing if a whitespace
         or empty string is passed
         """
-
-        self.instance.find_project("set", " ")
+        with self.assertRaises(MissingMandatoryParamError):
+            self.instance.find_project("set", "")
```
Comment (anish-mudaraddi) [inline tests/lib/test_openstack_identity.py:220]: This used to be passing `" "` - I'm assuming passing `""` is the same - just checking

File: lib/email_api/emailer.py (hunk 1 above)
Comment (DavidFair) [reply lib/email_api/emailer.py:83]: Originally I wanted to split it up, but I've reverted it and called it `build_email` (so we're not going in via protected)

**Comment by anish-mudaraddi** [comment]
closing this as it's been done
